### PR TITLE
Andrewbranch

### PR DESCRIPTION
**Pull request by Dufffyyy**

### CODE REVIEW

File: Snake/BusinessLayer/Engine.cs [hunk 1]
```diff
@@ -38,18 +38,21 @@ public enum snakeAction
         gameMode currentMode = gameMode.basic;
 
 
-        public Engine(int length, int width, int mode)
+        public Engine(int length = 20,  int width = 70,  int mode = 1)
```
Comment (shreyasn995) [inline Snake/BusinessLayer/Engine.cs]: These should be named constants

File: Snake/Snake/UILayer.cs [hunk 2]
```diff
@@ -91,12 +95,36 @@ static void endGame()
...
+            {
+                
+                Style.menuImage();
+                try
+                {
+                    gameMode = Convert.ToInt32(Console.ReadLine());
+                }
+                catch
+                {
+                    Console.WriteLine();
```
Comment (shreyasn995) [inline Snake/Snake/UILayer.cs]: Can you do something if you catch an exception?

**Comment by danaterr** [comment]
check the git ignore settings. this file: Snake/BusinessLayer/obj/Debug/DesignTimeResolveAssemblyReferencesInput.cache

looks like it should not be there.

File: Snake/Snake/UILayer.cs [hunk 3]
```diff
@@ -91,12 +95,36 @@ static void endGame()
             System.Console.WriteLine("Enter r to replay.");
 
             UserReplay = Console.ReadLine().ToString();
-            if (UserReplay == "R" || UserReplay == "r")
+            if (UserReplay.Equals("r", StringComparison.CurrentCultureIgnoreCase))
```
Comment (danaterr) [inline Snake/Snake/UILayer.cs]: use ordinalignorecase instead

File: Snake/Snake/UILayer.cs (hunk 2 above)
Comment (danaterr) [inline Snake/Snake/UILayer.cs]: what would you do if you caught it? seems like you are screwed anyway. Consider how a higher up global exception handler could help instead and don't try catch these lower down exceptions unless you can really do something about it at that lower level.

File: Snake/BusinessLayer/Engine.cs [hunk 4]
```diff
@@ -35,33 +38,29 @@ public enum snakeAction
             die,
             move
         }
-        
 
         gameMode currentMode = gameMode.basic;
 
-
-        public Engine(int length = mazeRenderLength,  int width = mazeRenderWidth,  int mode = 1)
+        public Engine(int length, int width, int mode)
```
Comment (wjoneil) [inline Snake/BusinessLayer/Engine.cs]: Looks like a bad merge - you've lost your changes here

**Pull request merged**